src/mbed_tools/project/mbed_program.py

```diff
@@ -113,6 +113,9 @@ def parse_url(name_or_url: str) -> Dict[str, str]:
     url_obj = urlparse(name_or_url)
     if url_obj.hostname:
         url = url_obj.geturl()
+    elif ":" in name_or_url.split("/", maxsplit=1)[0]:
```

Comment:
Does this cover all valid `git` urls? Should we try and use a library to parse the url instead? There are at least 2 libraries that claim to handle parsing `git` urls on pypi, although I have no idea about their quality.
https://github.com/git/git/blob/master/t/t0110-urlmatch-normalization.sh here are the test cases `git` uses for its url parser. I don't know how many of these formats we need to support or if your changes mean we now support all of them.

Reply:
From `git clone` docs, for URLs `git` supports either `scheme://` formats, which `urlparse` will parse, or the `[user@]host:` format, which is only recognized by `git` if there is a no slash before the first colon. Adding the check means that format will now be supported in addition to the cases `urlparse` covers. I did look at using an available parsing library, but it might be unnecessary if we're just passing the possible URLs to `git`.

Reply:
OK, seems fine then. Thanks for explaining.